### PR TITLE
Java mail sender

### PR DESCRIPTION
**Pull request by sameerstha0012**

### CODE REVIEW

File: src/main/java/com/theateam/waaprojectspringbackend/controller/SendMailerController.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                return ResponseEntity.ok("Emails sent successfully to multiple recipients");
+            } else {
+                return ResponseEntity.badRequest().body("No recipients provided");
+            }
+        } else {
+            return ResponseEntity.badRequest().body("Subject and body are required");
+        }
+    }
+
+    @PostMapping("/sendHtmlEmailWithAttachment")
```
Comment (vytaux) [inline src/main/java/com/theateam/waaprojectspringbackend/controller/SendMailerController.java:42]: We dont need any mail with attachments.

File: src/main/java/com/theateam/waaprojectspringbackend/service/EmailService.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    public void sendEmail(String subject, String body, List<String> recipients) {
+        SimpleMailMessage message = new SimpleMailMessage();
+        message.setSubject(subject);
+        message.setText(body);
+        message.setTo(recipients.toArray(new String[0]));
+        mailSender.send(message);
+    }
+
+    @Async
+    public void sendHtmlEmailWithAttachment(String subject, String body, List<String> recipients, Map<String, String> placeholders, MultipartFile attachment) throws MessagingException, IOException {
```
Comment (vytaux) [inline src/main/java/com/theateam/waaprojectspringbackend/service/EmailService.java]: we probably dont need this

File: src/main/java/com/theateam/waaprojectspringbackend/service/EmailService.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+@Service
+public class EmailService {
+    @Autowired
+    private JavaMailSender mailSender;
+
+    @Value("${spring.email.sender}")
+    private String senderEmail;
+
+    @Async
```
Comment (vytaux) [inline src/main/java/com/theateam/waaprojectspringbackend/service/EmailService.java:39]: We probably just need this one method? And use it where we need notifications, maybe you could use it in necessary places

File: src/main/java/com/theateam/waaprojectspringbackend/config/SecurityConfiguration.java [hunk 4]
```diff
@@ -41,6 +41,8 @@ protected void configure(HttpSecurity http) throws Exception {
                 .antMatchers("/api/v1/admin/**").hasAuthority(RoleType.ADMIN.name())
                 // Necessary for H2 console access
                 .antMatchers("/h2-console/**").permitAll()
+                .antMatchers("/api/sendMail").permitAll()
```
Comment (vytaux) [inline src/main/java/com/theateam/waaprojectspringbackend/config/SecurityConfiguration.java:44]: we probably dont need a separate controller

**Review by vytaux — CHANGES_REQUESTED** [review]

**Review by sanjeev-thapa — APPROVED** [review]

File: src/main/java/com/theateam/waaprojectspringbackend/controller/SendMailerController.java (hunk 1 above)
Comment (sameerstha0012) [reply src/main/java/com/theateam/waaprojectspringbackend/controller/SendMailerController.java:42]: Yeah, there is a method in service without attachment too

File: src/main/java/com/theateam/waaprojectspringbackend/service/EmailService.java (hunk 3 above)
Comment (sameerstha0012) [reply src/main/java/com/theateam/waaprojectspringbackend/service/EmailService.java:39]: yeah, thats it if we just send plain mail to user

**Pull request merged**